homeassistant/helpers/__init__.py

```diff
@@ -98,14 +98,17 @@ def config_per_platform(config, domain, logger):
 
     while config_key in config:
```

Comment:
If I get it correctly, this adds support to specify multiple platforms in the following format:
```yaml
switch:
  -
    platform: wemo
    password: bla
  -
    platform: wink
    option: bla
```

We can make it even a bit more succinct:

```yaml
switch:
  wemo:
    password: bla
  wink:
    option: bla
```

We could make that backwards compatible by testing for old style configurations: 
```python
if CONF_PLATFORM in platform_config:
  yield platform_config[CONF_PLATFORM], platform_config
else:
  try:
    yield from platform_config.items()
  except AttributeError:
    # It is not a dict
    break
```